### PR TITLE
[#536]  Prevent statements from going missing if updated in source CSV

### LINKED ISSUE #536: Valid statements can be hidden by mistake
Updating the data in a CSV source can cause statements to be hidden. Eg the original statement being marked as removed from source and the other new statement being marked as a duplicate.

The duplicate is detected before the `removed_from_source` flag is set. We should reverse this and and ensure we ignore removed statements when detecting duplicates. 

We need to update existing statements which are effected.

Problem statements can be seen by running:
```
SELECT `statements`.* FROM `statements` WHERE `statements`.`page_id` = 365 AND `statements`.`electoral_district_item` = 'Q55339185'
```

### PR DESCRIPTION
**Pull request by gbp**
Fixes #536

### CODE REVIEW

File: app/services/load_statements.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ def run
       touched_statements = csv.map { |result| parse_result(result) }
       untouched_statements = page.statements.where.not(id: touched_statements)
       untouched_statements.update(removed_from_source: true)
+      touched_statements.map(&:save!)
```
Comment (alexsdutton) [inline app/services/load_statements.rb]: Should this be `touched_statements.each(&:save!)`?

Comment (gbp) [reply app/services/load_statements.rb]: As were not using the result, yes. Good catch

File: spec/services/load_statements_spec.rb [hunk 2]
```diff
@@ -207,5 +207,31 @@
...
+
+      let!(:existing) { create(:statement, person_name: 'Alice', page: page) }
+      let(:statement) { Statement.last }
+
+      before do
+        load_statements = LoadStatements.new(page.title)
+        expect { load_statements.run }.to change(Statement, :count).by(1)
+      end
+
+      it 'should replaces existing statement' do
```
Comment (tmtmtmtm) [inline spec/services/load_statements_spec.rb]: typo: replaces → replace

File: spec/services/load_statements_spec.rb [hunk 3]
```diff
@@ -207,5 +207,31 @@
...
+        existing.reload
+        expect(existing).to be_removed_from_source
+      end
+
+      it 'should not be mark new statement as a duplicate' do
```
Comment (tmtmtmtm) [inline spec/services/load_statements_spec.rb]: typo: 'not be mark' → 'not mark'

**Pull request merged**